test/test_script.py

```diff
@@ -1174,6 +1180,30 @@ def test_cmd_datetime(self):
             # Restore original datetime object
             datetime.datetime = original_datetime
 
+    @skipUnless(IS_WIN, "windows test")
+    def test_cmd_datetime_windows(self):
+        ''' Test failure specific to Windows only. '''
+        # Save origninal datetime object and substitute one returning
+        # a fixed now() value for testing.
+        original_datetime = datetime.datetime
+        datetime.datetime = _DateTime
+
+        try:
+            context = Metadata()
+            # Tests with Windows-specific format
+            self.assertScriptResultEquals(r"$datetime(\%Y-\%#m-\%#d)", "2020-1-2", context)
+            # Tests with invalid format codes (Windows does not support -d or -m codes)
+            areg = r"^\$datetime:\d+:\d+: Unsupported format code"
+            with self.assertRaisesRegex(ScriptRuntimeError, areg):
```

Comment:
What about testing `ScriptRuntimeError` on non-win systems?

Reply:
I was going to, but I haven't yet found anything that would cause it to trigger.  Linux and Mac seem to be much more tolerant about the `strftime` format string.  Are you aware of something?

Reply:
I just tested on linux with Python 3.6.2: when `strftime()` is passed an unsupported format string (like `%E` or `%Q`) it is just returned as is (this behavior differs on Windows, where unsupported format raises exception). So you're right, at least on this implementation, there's no way to trigger this error this way.

I tested all ascii chars from 1 to 255:

```python
for i in range(1, 255):
    print("%s %s" % (chr(i), datetime.datetime.now(tz=local_tz).strftime("%" + chr(i))))
```
None of those leads to ValueError.

**BUT there's one case: if the format string ends with '%'**

```python
>>> datetime.datetime.now(tz=local_tz).strftime("%")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: strftime format ends with raw %
```

https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior